File: rex/scripts/Greet.java
public class Greet {
    private String who;

    public Greet(String who) {
        this.who = who;
    }

    public void salute() {
        System.out.println("Hello " +
            who.toUpperCase().charAt(0) +
            who.substring(1) + "!");
    }

    public static void main(String... args) {
        final Greet g = new Greet("world");
        g.salute();
    }
}

File: js/greet.js
function Greet(who) {
    this.name = who[0].toUpperCase() +
        who.substr(1);
}

Greet.prototype.salute = function() {
    console.log("Hello " + this.name + "!");
};

var g = new Greet("world"); // create object
g.salute(); // output "Hello World!"